### PR TITLE
[Monitor] BREAKING CHANGE: `az monitor activity-log alert create`: Change default value from resourceGroupId to subscriptionId for parameter `--scope`

### PR DESCRIPTION
**Pull request by AllyW**
**Related command**

az monitor activity-log list
az monitor activity-log list-categories
az monitor activity-log alert CRUD
az monitor activity-log alert action-group
az monitor activity-log alert scope
**Description**

**Testing Guide**

**History Notes**

[Monitor] BREAKING CHANGE: `az monitor activity-log alert create`: Change default value from resourceGroupId to subscriptionId for parameter `--scope`
[Monitor] `az monitor activity-log`: Migrate commands to codegen V2

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
monitor

File: src/azure-cli/azure/cli/command_modules/monitor/operations/activity_log_alerts.py [hunk 1]
```diff
@@ -2,192 +2,464 @@
...
+            args.all_of = [{
+                "field": "category",
+                "equals": "ServiceHealth",
+            }]
+        # Add action groups
+        action_group_rids = set()
+        if has_value(args.action_group_id):
+            action_group_rids = _normalize_names_instance(self.cli_ctx, args.action_group_id.to_serialized_data(),
+                                                          args.resource_group.to_serialized_data(),
+                                                          'microsoft.insights', 'actionGroups')
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/monitor/operations/activity_log_alerts.py:173]: Is it `Microsoft.Insights` or `microsoft.insights`?

File: src/azure-cli/azure/cli/command_modules/monitor/operations/activity_log_alerts.py [hunk 2]
```diff
@@ -2,192 +2,464 @@
...
+                                                                                        args.resource_group))
+        if not has_value(args.all_of):
+            args.all_of = [{
+                "field": "category",
+                "equals": "ServiceHealth",
+            }]
+        # Add action groups
+        action_group_rids = set()
+        if has_value(args.action_group_id):
+            action_group_rids = _normalize_names_instance(self.cli_ctx, args.action_group_id.to_serialized_data(),
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/monitor/operations/activity_log_alerts.py:171]: Why not use AAZResourceIdArgFormat? Like this: https://github.com/Azure/azure-cli/blob/759bb130cc6736c80aab945a0902d13a75ce7b8f/src/azure-cli/azure/cli/command_modules/network/operations/nat.py#L18-L26

File: src/azure-cli/azure/cli/command_modules/monitor/operations/activity_log_alerts.py [hunk 3]
```diff
@@ -2,192 +2,464 @@
...
+                                                          args.resource_group.to_serialized_data(),
+                                                          'microsoft.insights', 'actionGroups')
+
+        args.action_groups = []
+        for i in action_group_rids:
+            args.action_groups.append({
+                "action_group_id": i,
+                "webhook_properties_raw": webhook_properties
+            })
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/monitor/operations/activity_log_alerts.py:180]: You should also use assign_aaz_list_arg for list transformation https://github.com/Azure/azure-cli/blob/759bb130cc6736c80aab945a0902d13a75ce7b8f/src/azure-cli/azure/cli/command_modules/network/operations/nat.py#L45-L49

File: src/azure-cli/azure/cli/command_modules/monitor/operations/activity_log_alerts.py [hunk 4]
```diff
@@ -2,192 +2,464 @@
...
+        return args_schema
+
+    def pre_instance_update(self, instance):
+        args = self.ctx.args
+        webhook_properties = process_webhook_properties(args)
 
-    return client.list_by_subscription_id()
+        # normalize the action group ids
+        rids = _normalize_names(self.cli_ctx, args.action_group_ids.to_serialized_data(), args.resource_group,
+                                'microsoft.insights', 'actionGroups')
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/monitor/operations/activity_log_alerts.py:282]: Same concern as above

File: src/azure-cli/azure/cli/command_modules/monitor/operations/activity_log_alerts.py (hunk 1 above)
Comment (AllyW) [reply src/azure-cli/azure/cli/command_modules/monitor/operations/activity_log_alerts.py:173]: In previous version, it is microsoft.insights here.

image

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**